modules/hashi-snapshots.nix

```
@@ -0,0 +1,416 @@
+{
+  config,
+  pkgs,
+  lib,
+  hashiTokens,
+  ...
+}: let
+  cfg = config.services.hashi-snapshots;
+
+  inherit (lib) boolToString listToAttrs mkEnableOption mkIf mkMerge mkOption nameValuePair toUpper;
+  inherit (lib.types) addCheck attrs bool enum int str submodule;
+
+  snapshotJobConfig = submodule {
+    options = {
+      enable = mkOption {
+        type = bool;
+        default = true;
+        description = ''
+          Creates a systemd service and timer to automatically save Vault snapshots.
+        '';
+      };
+
+      backupCount = mkOption {
+        type = addCheck int (x: x >= 0);
+        default = null;
+        description = ''
+          The number of snapshots to keep.  A sensible value matched to the onCalendar
+          interval parameter should be used.  Examples of sensible suggestions may be:
+
+            48 backupCount for "hourly" interval (2 days of backups)
+            30 backupCount for "daily" interval (1 month of backups)
+        '';
+      };
+
+      backupDirPrefix = mkOption {
+        type = str;
+        default = null;
+        description = ''
+          The top level location to store the snapshots.  The actual storage location
+          of the files will be this prefix path with the snapshot job name appended,
+          where the job is one of "hourly", "daily" or "custom".
+
+          Therefore, saved snapshot files will be found at:
+
+            $backupDirPrefix/$job/*.snap
+        '';
+      };
+
+      backupSuffix = mkOption {
+        type = addCheck str (x: x != "");
```

```suggestion
        type = nonEmptyStr;
```

The [check](https://github.com/NixOS/nixpkgs/blob/3bbb296d9a0088c314ce83038b896753bbe33acb/lib/types.nix#L311) is a bit different though. Same in line 103.